src/main/java/com/fasterxml/jackson/databind/deser/impl/PropertyValueBuffer.java

```diff
@@ -335,7 +335,12 @@ public Object handleIdValue(final DeserializationContext ctxt, Object bean) thro
                 // also: may need to set a property value as well
                 SettableBeanProperty idProp = _objectIdReader.idProperty;
                 if (idProp != null) {
-                    return idProp.setAndReturn(bean, _idValue);
+                    // [databind#5328] Records do not have setters, skip...to set id value
+                    if (bean.getClass().isRecord()) {
```

Comment:
Very much open for suggestion, couldn't think of more static way to get this done (unless we add property in `SettableBeanProperty`?